var/spack/repos/builtin/packages/mpich/package.py

```diff
@@ -90,6 +90,9 @@ class Mpich(AutotoolsPackage):
 '-Wl,-flat_namespace'.'''
     )
 
+    variant('max_vcis', values=int, default=1, when='device=ch4',
+            description='Select max number of VCIs to configure')
```

Comment:
The `--with-max-vcis` option does support `--with-max-vcis=default` so that you can get the right value for whatever version you are building.

Reply:
Ok so I set the  `--enable-thread-cs=per-vci` lflag when max_vci is set and I added a 'default' option that is used by default in case the version is superior of 3.4. 

I also added all possible configurations of `max_vcis` between 1 and 64 for the user. Should I reduce to a subset? For example, `('1', '8', '16', '32', '64')`.

Just for context, my objective here is to use 64 VCIs with the MPICH 3.4.2.

Reply:
There is no restriction on any number of VCIs between 1 and 64, though a power of 2 will likely generate better code when manipulating vci values. I think what you have now should work.

FWIW, you're probably less likely to run into bugs or thread contention issues using 4.0b1 with multiple VCIs. We should be putting out a stable release in the 4.0 series in the next few weeks regardless.

Reply:
One last note is that the number of VCIs used at *runtime* still defaults to 1. In order to actually use multiple VCIs, you need to set an envvar e.g.
```
    MPIR_CVAR_CH4_NUM_VCIS=<N>
```

Reply:
Understood, thanks!! 
We will probably move to 4.0 once the stable is out.